lib/core/commands/user-commands.js

```diff
@@ -7,31 +7,51 @@ class UserCommands {
   async createUser(user) {
     const command = {
       createUser: user.user,
-      // mongo internally uses pwd
-      pwd: user.password,
       customData: user.customData || {},
       roles: user.roles || []
     }
+
+    // mongo internally uses pwd
+    if (user.password) {
+      command.pwd = user.password
+    }
+    // Use external db iff external is set
+    if (user.external) {
+      this.client.useDatabase('$external')
+    }
+
     return await this.client.command(command)
   }
 
   // https://docs.mongodb.com/manual/reference/command/updateUser/
   async updateUser(user) {
     const command = {
       updateUser: user.user,
-      // mongo internally uses pwd
-      pwd: user.password,
       customData: user.customData || {},
       roles: user.roles || []
     }
+
+    // mongo internally uses pwd
+    if (user.password) {
+      command.pwd = user.password
+    }
+    // Use external db iff external is set
+    if (user.external) {
```

Comment:
While in technical terms `external` is the correct term, we could use something more clear to users like `certificate`

Reply:
No, because you use the same `external` syntax for other external auth providers like LDAP  and other services or custom auth.

https://docs.mongodb.com/manual/tutorial/authenticate-nativeldap-activedirectory/#user-objects 
https://docs.mongodb.com/manual/core/security-ldap-external/#mongodb-roles-for-ldap-authorization

The reason this is named as `$external` it to provide a clear model that mongo will look to an outside system/service to validate the users. Instead of purely username/password handled internally.